### DISCUSSION TITLE
Rewrite function API path

### DISCUSSION THREAD

**Discussion by juanmarinballastlane** [gatsbyjs/gatsby#34231] [Help] [answered]
Hello guys

There is a way to rewrite the api path to another? I mean, I have `/api/hello-world` and I want `hello-world`.

Thanks

**Comment by violy** [comment, marked as answer]
What are you trying to do? There's no API path on Gatsby default starters…

**Reply by juanmarinballastlane** [reply]
Thanks for review my comment, I want to do something like this https://docs.wpvip.com/technical-references/node-js/#h-requirement-1-exposing-a-health-check-endpoint.

**Reply by violy** [reply]
You want your Gatsby site expose a _site-health-like_ endpoint ? that's right ? 

The solution depends which *site health* you are monitoring … and so it depends on your hosting stack. is your stack pure SSG, or are you using SSR or DSG ?

**Reply by juanmarinballastlane** [reply]
yup, basically, WordpressVIP people needs to make some checks using the site url `www.domain.com/cache-healthcheck?` and that route should to return a 200 status. So, right now is working as `www.domain.com/api/cache-healthcheck?` and I need to remove /api from path

**Reply by violy** [reply]
first, have you simply try to put a `cache-healthcheck` in `static/cache-healthcheck` ?
If i do that, _locally_, and then, `build` & `serve` gatsby, I've that → 

```
curl -I http://localhost:9000/cache-healthcheck
HTTP/1.1 200 OK
X-Powered-By: Express
Access-Control-Allow-Origin: *
Access-Control-Allow-Headers: Origin, X-Requested-With, Content-Type, Accept
Accept-Ranges: bytes
Cache-Control: public, max-age=0
Last-Modified: Mon, 13 Dec 2021 21:04:39 GMT
ETag: W/"0-17db59c9b34"
Content-Type: application/octet-stream
Content-Length: 0
Date: Mon, 13 Dec 2021 21:05:19 GMT
Connection: keep-alive
Keep-Alive: timeout=5
```

it's a static `200` zero-length `application/octet-stream`. it might be suffisant ?
the response depend of your hosting stack. 

note 1 : according to [this line](https://github.com/gatsbyjs/gatsby/blob/dd46c640fc02ea0b3dfa8788ca4a3d21e7783fe4/packages/gatsby/src/commands/serve.ts#L138) on `gatsby serve` the middleware match only the `/api/*` path and it's hard-coded, highly opinionated. 

note 2 : depending on how is hosted your site (and your real need) you can find other ways (like _Netlify Function_)
